src/redux/books/books.js

```diff
@@ -38,29 +21,76 @@ export default function booksReducer(state = defaultState, action) {
           id: action.id,
           title: action.title,
           author: action.author,
-          progress: action.progress,
-          currentChapter: action.currentChapter,
+          category: action.category,
         },
       ];
     case REMOVE:
       return state.filter((book) => book.id !== action.id);
+    case FETCH:
+      return action.list;
     default:
       return state;
   }
 }
 
-// action creators for add and remove books
+// action creators for fetch, add and remove books
+
+export const fetchBook = (list) => ({ type: FETCH, list });
+
+export const fetchBookAsync = () => async (dispatch) => {
```

Comment:
- [ ] This functionality should be created using the `createAsyncThunk`, kindly consider making the changes so your thunk action creators use `createAsyncThunk` instead

Reply:
@Ismailco Hi! The function uses an async thunk (a function that returns another function), i am just not using the words createAsyncThunk.

Can you consider this version to be okay?

Reply:
@MattGomb, yes I know that the requirement specifically mentioned using the `createAsyncThunk`, kindly make sure your code uses this method to create the thunk action creator